### ISSUE TITLE
Test fails to compile on 32-bit architectures

### ISSUE THREAD

**Issue by siretart** [cel-expr/cel-go#971]
Hi, I'm packaging this library for use of other packages in Debian.

I noticed that in Debian, the tests fails to build on 32bit architectures, e.g.:

- armel: https://ci.debian.net/packages/g/golang-github-google-cel-go/testing/armel/48129938/
- armef: https://ci.debian.net/packages/g/golang-github-google-cel-go/testing/armhf/48129983/
- i386: https://ci.debian.net/packages/g/golang-github-google-cel-go/testing/i386/48130026/

In all cases, the relevant part of the build log looks like this:

```
132s # github.com/google/cel-go/common/types [github.com/google/cel-go/common/types.test]
132s src/github.com/google/cel-go/common/types/int_test.go:41:17: cannot use math.MaxInt64 (untyped int constant 9223372036854775807) as int value in assignment (overflows)
132s src/github.com/google/cel-go/common/types/int_test.go:44:17: cannot use math.MinInt64 (untyped int constant -9223372036854775808) as int value in assignment (overflows)
132s src/github.com/google/cel-go/common/types/int_test.go:47:17: cannot use math.MaxInt64 - 1 (untyped int constant 9223372036854775806) as int value in assignment (overflows)
132s src/github.com/google/cel-go/common/types/int_test.go:48:63: cannot use math.MaxInt64 (untyped int constant 9223372036854775807) as int value in argument to t.Errorf (overflows)
132s src/github.com/google/cel-go/common/types/int_test.go:50:17: cannot use math.MinInt64 + 1 (untyped int constant -9223372036854775807) as int value in assignment (overflows)
132s src/github.com/google/cel-go/common/types/int_test.go:51:63: cannot use math.MaxInt64 (untyped int constant 9223372036854775807) as int value in argument to t.Errorf (overflows)
132s src/github.com/google/cel-go/common/types/int_test.go:355:17: cannot use math.MinInt64 (untyped int constant -9223372036854775808) as int value in assignment (overflows)
132s src/github.com/google/cel-go/common/types/int_test.go:420:17: cannot use math.MinInt64 (untyped int constant -9223372036854775808) as int value in assignment (overflows)
132s src/github.com/google/cel-go/common/types/int_test.go:432:17: cannot use math.MaxInt64 / 2 (untyped int constant 4611686018427387903) as int value in assignment (overflows)
132s src/github.com/google/cel-go/common/types/int_test.go:435:17: cannot use math.MinInt64 / 2 (untyped int constant -4611686018427387904) as int value in assignment (overflows)
132s src/github.com/google/cel-go/common/types/int_test.go:435:17: too many errors
```

I suspect the issue is here:
https://github.com/google/cel-go/blob/b27008a343bcb5805b63816e41954b3100a05be5/common/types/int_test.go#L35-L54

That test code seems to be written with the assumption to run on 64bit architectures only. I'm going to disable this test in Debian for now, but please let me know if you can think of a better way to address this issue.

**Comment by TristonianJones** [comment]
My apologies @siretart, I thought I had replied to this thread, but it's possible there are other issues in `common/types/int.go` and `common/types/overflow.go`. I'm happy to update reference to these constants where they occur. If I read this right it means type converting the constants, e.g. `int64(math.MaxInt64)`, does that sound right to you?

**Comment by TristonianJones** [comment]
I've successfully compiled 32-bit builds for windows and linux, but wasn't able to access the OS and architecture used in this PR. 

```sh
GOOS=linux GOARCH=386 go build github.com/google/cel-go
```

I'm going to close this PR for now. Please feel free to re-open if you have any issues with the build targets available from `go tool dist list`

**Comment by siretart** [comment]
the issue is in the test, not in the build. Please try this:

```
GOOS=linux GOARCH=386 go test  github.com/google/cel-go
```

**Comment by siretart** [comment]
this seems to fix it:

```
commit a46e95a0a336baecaf9357ed6b592a0cfabf36f9
Author: Reinhard Tartler <siretart@gmail.com>
Date:   Fri Mar 27 20:53:55 2026 +0000

    fix: update integer test cases to use int64 for boundary conditions
    
    Fixes: #971
    Signed-off-by: Reinhard Tartler <siretart@gmail.com>

diff --git a/common/types/int_test.go b/common/types/int_test.go
index 560c3df..2bf2d73 100644
--- a/common/types/int_test.go
+++ b/common/types/int_test.go
@@ -39,17 +39,17 @@ func TestIntAdd(t *testing.T) {
 	if !IsError(Int(-1).Add(String("-1"))) {
 		t.Error("Adding non-int to int was not an error.")
 	}
-	if lhs, rhs := math.MaxInt64, 1; !IsError(Int(lhs).Add(Int(rhs))) {
+	if lhs, rhs := int64(math.MaxInt64), int64(1); !IsError(Int(lhs).Add(Int(rhs))) {
 		t.Errorf("Expected adding %d and %d to result in overflow.", lhs, rhs)
 	}
-	if lhs, rhs := math.MinInt64, -1; !IsError(Int(lhs).Add(Int(rhs))) {
+	if lhs, rhs := int64(math.MinInt64), int64(-1); !IsError(Int(lhs).Add(Int(rhs))) {
 		t.Errorf("Expected adding %d and %d to result in overflow.", lhs, rhs)
 	}
-	if lhs, rhs := math.MaxInt64-1, 1; !Int(lhs).Add(Int(rhs)).Equal(Int(math.MaxInt64)).(Bool) {
-		t.Errorf("Expected adding %d and %d to yield %d", lhs, rhs, math.MaxInt64)
+	if lhs, rhs := int64(math.MaxInt64-1), int64(1); !Int(lhs).Add(Int(rhs)).Equal(Int(int64(math.MaxInt64))).(Bool) {
+		t.Errorf("Expected adding %d and %d to yield %d", lhs, rhs, int64(math.MaxInt64))
 	}
-	if lhs, rhs := math.MinInt64+1, -1; !Int(lhs).Add(Int(rhs)).Equal(Int(math.MinInt64)).(Bool) {
-		t.Errorf("Expected adding %d and %d to yield %d", lhs, rhs, math.MaxInt64)
+	if lhs, rhs := int64(math.MinInt64+1), int64(-1); !Int(lhs).Add(Int(rhs)).Equal(Int(int64(math.MinInt64))).(Bool) {
+		t.Errorf("Expected adding %d and %d to yield %d", lhs, rhs, int64(math.MinInt64))
 	}
 }
 
@@ -383,7 +383,7 @@ func TestIntDivide(t *testing.T) {
 	if !IsError(Int(1).Divide(Double(-1))) {
 		t.Error("Division permitted without express type-conversion.")
 	}
-	if lhs, rhs := math.MinInt64, -1; !IsError(Int(lhs).Divide(Int(rhs))) {
+	if lhs, rhs := int64(math.MinInt64), int64(-1); !IsError(Int(lhs).Divide(Int(rhs))) {
 		t.Errorf("Expected dividing %d and %d result in overflow.", lhs, rhs)
 	}
 }
@@ -462,7 +462,7 @@ func TestIntModulo(t *testing.T) {
 	if !IsError(Int(21).Modulo(uintZero)) {
 		t.Error("Modulus permitted between different types without type conversion.")
 	}
-	if lhs, rhs := math.MinInt64, -1; !IsError(Int(lhs).Modulo(Int(rhs))) {
+	if lhs, rhs := int64(math.MinInt64), int64(-1); !IsError(Int(lhs).Modulo(Int(rhs))) {
 		t.Errorf("Expected modulo %d and %d result in overflow.", lhs, rhs)
 	}
 }
@@ -474,25 +474,25 @@ func TestIntMultiply(t *testing.T) {
 	if !IsError(Int(1).Multiply(Double(-4.0))) {
 		t.Error("Multiplication permitted without express type-conversion.")
 	}
-	if lhs, rhs := math.MaxInt64/2, 3; !IsError(Int(lhs).Multiply(Int(rhs))) {
+	if lhs, rhs := int64(math.MaxInt64/2), int64(3); !IsError(Int(lhs).Multiply(Int(rhs))) {
 		t.Errorf("Expected multiplying %d and %d to result in overflow.", lhs, rhs)
 	}
-	if lhs, rhs := math.MinInt64/2, 3; !IsError(Int(lhs).Multiply(Int(rhs))) {
+	if lhs, rhs := int64(math.MinInt64/2), int64(3); !IsError(Int(lhs).Multiply(Int(rhs))) {
 		t.Errorf("Expected multiplying %d and %d to result in overflow.", lhs, rhs)
 	}
-	if lhs, rhs := math.MaxInt64/2, 2; !Int(lhs).Multiply(Int(rhs)).Equal(Int(math.MaxInt64 - 1)).(Bool) {
-		t.Errorf("Expected multiplying %d and %d to yield %d", lhs, rhs, math.MaxInt64-1)
+	if lhs, rhs := int64(math.MaxInt64/2), int64(2); !Int(lhs).Multiply(Int(rhs)).Equal(Int(int64(math.MaxInt64 - 1))).(Bool) {
+		t.Errorf("Expected multiplying %d and %d to yield %d", lhs, rhs, int64(math.MaxInt64-1))
 	}
-	if lhs, rhs := math.MinInt64/2, 2; !Int(lhs).Multiply(Int(rhs)).Equal(Int(math.MinInt64)).(Bool) {
-		t.Errorf("Expected multiplying %d and %d to yield %d", lhs, rhs, math.MinInt64)
+	if lhs, rhs := int64(math.MinInt64/2), int64(2); !Int(lhs).Multiply(Int(rhs)).Equal(Int(int64(math.MinInt64))).(Bool) {
+		t.Errorf("Expected multiplying %d and %d to yield %d", lhs, rhs, int64(math.MinInt64))
 	}
-	if lhs, rhs := math.MaxInt64/2, -2; !Int(lhs).Multiply(Int(rhs)).Equal(Int(math.MinInt64 + 2)).(Bool) {
-		t.Errorf("Expected multiplying %d and %d to yield %d", lhs, rhs, math.MinInt64+2)
+	if lhs, rhs := int64(math.MaxInt64/2), int64(-2); !Int(lhs).Multiply(Int(rhs)).Equal(Int(int64(math.MinInt64 + 2))).(Bool) {
+		t.Errorf("Expected multiplying %d and %d to yield %d", lhs, rhs, int64(math.MinInt64+2))
 	}
-	if lhs, rhs := (math.MinInt64+2)/2, -2; !Int(lhs).Multiply(Int(rhs)).Equal(Int(math.MaxInt64 - 1)).(Bool) {
-		t.Errorf("Expected multiplying %d and %d to yield %d", lhs, rhs, math.MaxInt64-1)
+	if lhs, rhs := int64((math.MinInt64+2)/2), int64(-2); !Int(lhs).Multiply(Int(rhs)).Equal(Int(int64(math.MaxInt64 - 1))).(Bool) {
+		t.Errorf("Expected multiplying %d and %d to yield %d", lhs, rhs, int64(math.MaxInt64-1))
 	}
-	if lhs, rhs := math.MinInt64, -1; !IsError(Int(lhs).Multiply(Int(rhs))) {
+	if lhs, rhs := int64(math.MinInt64), int64(-1); !IsError(Int(lhs).Multiply(Int(rhs))) {
 		t.Errorf("Expected multiplying %d and %d result in overflow.", lhs, rhs)
 	}
 }
@@ -501,11 +501,11 @@ func TestIntNegate(t *testing.T) {
 	if !Int(1).Negate().Equal(Int(-1)).(Bool) {
 		t.Error("Negating int value did not succeed")
 	}
-	if v := math.MinInt64; !IsError(Int(v).Negate()) {
+	if v := int64(math.MinInt64); !IsError(Int(v).Negate()) {
 		t.Errorf("Expected negating %d to result in overflow.", v)
 	}
-	if v := math.MaxInt64; !Int(v).Negate().Equal(Int(math.MinInt64 + 1)).(Bool) {
-		t.Errorf("Expected negating %d to yield %d", v, math.MinInt64+1)
+	if v := int64(math.MaxInt64); !Int(v).Negate().Equal(Int(int64(math.MinInt64 + 1))).(Bool) {
+		t.Errorf("Expected negating %d to yield %d", v, int64(math.MinInt64+1))
 	}
 }
 
@@ -516,16 +516,16 @@ func TestIntSubtract(t *testing.T) {
 	if !IsError(Int(1).Subtract(Uint(1))) {
 		t.Error("Subtraction permitted without express type-conversion.")
 	}
-	if lhs, rhs := math.MaxInt64, -1; !IsError(Int(lhs).Subtract(Int(rhs))) {
+	if lhs, rhs := int64(math.MaxInt64), int64(-1); !IsError(Int(lhs).Subtract(Int(rhs))) {
 		t.Errorf("Expected subtracting %d and %d to result in overflow.", lhs, rhs)
 	}
-	if lhs, rhs := math.MinInt64, 1; !IsError(Int(lhs).Subtract(Int(rhs))) {
+	if lhs, rhs := int64(math.MinInt64), int64(1); !IsError(Int(lhs).Subtract(Int(rhs))) {
 		t.Errorf("Expected subtracting %d and %d to result in overflow.", lhs, rhs)
 	}
-	if lhs, rhs := math.MaxInt64-1, -1; !Int(lhs).Subtract(Int(rhs)).Equal(Int(math.MaxInt64)).(Bool) {
-		t.Errorf("Expected subtracting %d and %d to yield %d", lhs, rhs, math.MaxInt64)
+	if lhs, rhs := int64(math.MaxInt64-1), int64(-1); !Int(lhs).Subtract(Int(rhs)).Equal(Int(int64(math.MaxInt64))).(Bool) {
+		t.Errorf("Expected subtracting %d and %d to yield %d", lhs, rhs, int64(math.MaxInt64))
 	}
-	if lhs, rhs := math.MinInt64+1, 1; !Int(lhs).Subtract(Int(rhs)).Equal(Int(math.MinInt64)).(Bool) {
-		t.Errorf("Expected subtracting %d and %d to yield %d", lhs, rhs, math.MinInt64)
+	if lhs, rhs := int64(math.MinInt64+1), int64(1); !Int(lhs).Subtract(Int(rhs)).Equal(Int(int64(math.MinInt64))).(Bool) {
+		t.Errorf("Expected subtracting %d and %d to yield %d", lhs, rhs, int64(math.MinInt64))
 	}
 }
```
